tests/net/ipv6/src/main.c

```diff
@@ -795,11 +809,61 @@ ZTEST(net_ipv6, test_send_neighbor_discovery)
 	zassert_equal(net_ipv6_nbr_data(nbr)->state, NET_IPV6_NBR_STATE_REACHABLE,
 		      "Neighbor should be reachable at this point.");
 
-	/* Second attempt (neighbor valid) should give no NS. */
+	/* Packet count should now be 3, one for the first NS and two
+	 * for the queued packets.
+	 */
+	zassert_equal(pkt_num, 3, "Unexpected number of packets sent (%d)", pkt_num);
+
+	/* Third attempt (neighbor valid) should give no NS. */
 	verdict = send_msg(&my_addr, &test_router_addr);
 	zassert_equal(verdict, NET_OK, "Packet was dropped (%d)", verdict);
 	zassert_equal(k_sem_take(&ctx.wait_ns, K_MSEC(10)), -EAGAIN,
 		      "Should not get NS");
+
+	/* Packet count should be 4 as we sent one more packet. */
+	zassert_equal(pkt_num, 4, "Unexpected number of packets sent (%d)", pkt_num);
+}
```

Comment:
Note to self to add tests that make sure ref counts are set correctly and `net_pkt` is removed after successfull TX.